002_simulation_structure/002_simulation_structure.tex

```diff
@@ -17,8 +17,205 @@ \section{Simulation Flow}\label{simulation_flow}
 %%%%%%%% Message Passing
 %%%%%%%%
 \section{Message Passing}\label{message_passing}
-Need a nice diagram for this
+\subsection{Primitive Messaging}
+An important role of Team 1 was to design a mechanism for agents to talk amongst each other in the simulation that would allow automated communication. The first step in this process was to design a system which would allow an agent to pass along a minimum viable message to their neighbours. These were the goals of communication in the MVP:
+\begin{itemize}
+    \item Agents could only talk to their immediate neighbors (+/- 1 floor).
+    \item Agents should be given the ability to ignore communication.
+    \item Multiple agents could be “speaking” at a given tick.
+    \item One agent could send one message per tick.
+    \item \textbf{It is entirely an agent's discretion if/when/how they wish to react to a message as long as their behavior is not deceptive}
+\end{itemize}
+As a team, Team 1 emphasized the last point. The infrastructure should not force any behavior onto the agent as long as the behavior was honest. The basic philosophy of the communication infrastructure was to dictate as little as possible about the agent's behavior while keeping the API friendly, readable and hard to get wrong. This usually meant making providing basic override-able behaviors to be as unobtrusive as possible to team strategies. \newline
+The minimum default message had to have the following fields that are accessible to all the agents that receive it:
+\begin{itemize}
+    \item \textbf{\texttt{MessageType}}: Indicates one of 13 enumerated message types.
```

Comment:
```suggestion
    \item \textbf{\texttt{MessageType}}: Indicates the message type.
```

Just flagging this for inconsistency. We mention the need for a common language between agents later but here we define the message fields as they were after the common language had been implemented. I'm happy for this to not change if people do no see as an issue.